core/src/rpc.rs

```diff
@@ -421,21 +402,33 @@ impl JsonRpcRequestProcessor {
         &self,
         signatures: Vec<Signature>,
         commitment: Option<CommitmentConfig>,
-    ) -> Result<Vec<Option<RpcTransactionStatus>>> {
+    ) -> RpcResponse<Vec<Option<RpcTransactionStatus>>> {
         let mut statuses: Vec<Option<RpcTransactionStatus>> = vec![];
 
         let bank = self.bank(commitment);
 
         for signature in signatures {
             let status = bank.get_signature_confirmation_status(&signature).map(
-                |SignatureConfirmationStatus { slot, status, .. }| RpcTransactionStatus {
+                |SignatureConfirmationStatus {
+                     slot,
+                     status,
+                     confirmations,
+                 }| RpcTransactionStatus {
                     slot,
                     status,
+                    confirmations: if confirmations < MAX_LOCKOUT_HISTORY {
```

Comment:
My mistake, I believe this should be `<= MAX_LOCKOUT_HISTORY`.

Reasoning:
* A slot is considered to be rooted once a node has voted for `MAX_LOCKOUT_HISTORY` (31) slots since the original slot.
* Confirmations are counted starting with the slot in which a signature's corresponding tx was processed
* So, a signature is in a rooted slot when it has 32 confirmations
* Therefore, anything less than 32 (`<= 31`) is not rooted

I think we should have a const for `MAX_CONFIRMATIONS` which is equal to `MAX_LOCKOUT_HISTORY + 1`, thoughts?